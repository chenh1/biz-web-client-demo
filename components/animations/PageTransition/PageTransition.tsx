'use client'

import React, { useEffect, useRef, useState } from 'react';
import { usePathname } from "next/navigation";
import { Heading } from '../../core/Heading/Heading';

const useNavigationEvent = (onPathnameChange) => {
  const pathname = usePathname(); // Get current route

  // Save pathname on component mount into a REF
  const savedPathNameRef = useRef(pathname);

  useEffect(() => {
    // If REF has been changed, do the stuff
    if (savedPathNameRef.current !== pathname) {
      onPathnameChange();
      // Update REF
      savedPathNameRef.current = pathname;
    }
  }, [pathname, onPathnameChange]);
};

export const PageTransition: React.FC<{}> = () => {
  const [ startTransition, setStartTransition ] = useState(false);
  //useNavigationEvent(() => setStartTransition(true));

  useEffect(() => {
    //if (startTransition) setTimeout(() => setStartTransition(false), 5000)
  }, [startTransition, setStartTransition])

  return startTransition ?
    <div className="fixed w-screen h-screen bg-apex-blue opacity-50 place-items-center">
      <Heading type="h4">TRANSITIONING</Heading>
    </div>
    : <></>
};