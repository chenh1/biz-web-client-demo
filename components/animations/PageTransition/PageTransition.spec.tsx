import { PageTransition } from "./PageTransition";
import { render, waitFor } from '@testing-library/react';
import { JSDOM } from "jsdom"
const dom = new JSDOM()
global.document = dom.window.document
global.window = dom.window


it('transitions pages via fade', async () => {
  const { getByTestId, asFragment } = render(
    <PageTransition/>
  );

  await waitFor(() => {
    expect(asFragment()).toMatchSnapshot();
  });
});