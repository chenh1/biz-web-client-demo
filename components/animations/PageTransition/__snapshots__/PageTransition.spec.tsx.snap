// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transitions pages via fade 1`] = `<DocumentFragment />`;
